perf/perf_pcp.py

```diff
@@ -89,12 +89,14 @@ def test_perfevent_24x7_events(self):
         self.lpar_24x7_ppc64le_check()
         if self.cpu_family == 'power8':
             self.cancel("Not supported on Power8")
-        # -s 2 collects two samples
-        cmd = "pmval -s 2 perfevent.hwcounters.hv_24x7.PM_MBA0_CLK_CYC.value"
-        process.run(cmd, shell=True)
-        # -s 2 collects two samples
-        cmd = "pmval -s 2 perfevent.hwcounters.hv_24x7.PM_PB_CYC.value"
-        process.run(cmd, shell=True)
+        output = process.system_output("pminfo").decode()
+        for line in output.split('\n'):
+            # -s 2 collects two samples
+            if 'PM_MBA0_CLK_CYC' in line and 'value' in line:
+                process.run('pmval -s 2 ' + line, shell=True)
+            # -s 2 collects two samples
+            if 'PM_PB_CYC' in line and 'value' in line:
+                process.run('pmval -s 2 ' + line, shell=True)
```

Comment:
If you are sure `PM_PB_CYC` line will follow after `PM_MBA0_CLK_CYC` line, you can add a `break` next to this line so that unnecessary looping can be avoided.